### DISCUSSION TITLE
Oracle Cloud integration?

### DISCUSSION THREAD

**Discussion by jacobsteffenBYU** [Budibase/budibase#10200] [Ideas]
I know that an Oracle integration exists, but are there any plans to support Oracle Cloud DB connections? I have a database I was trying to connect with TLS and Oracle cloud lists that in order to connect:

You are using JDBC Thin Client (version 12.2.0.1 or higher) with JDK 8(u163+) or higher.
You are using the Python python-oracledb driver.
You are using ODP.NET version 19.14 (or higher), or 21.5 (or higher).
You are using an Oracle Call Interface based driver with Oracle Client libraries version 19.14 (or higher), or 21.5 (or higher).

There's more info here as well: https://docs.oracle.com/en-us/iaas/autonomous-database-shared/doc/connect-introduction.html

**Comment by ConorWebb96** [comment]
Hey @jacobsteffenBYU,

Currently this functionality isn't in Budibase. I did find a potential workaround discussed on discord: https://discord.com/channels/733030666647765003/1024684188760293508/1080935497112760362 

If you don't have access to discord i've pasted the potential fix below:

> Hi @MegaDog¬ , I looked into this again. I was able to connect to the Oracle Cloud database using the Oracle tools 'sqlplus' and 'SQL Explorer' but to do so I needed to add 'wallet_location' to the connection string like this:
```sqlplus -l "ADMIN@tcps://adb.uk-london-1.oraclecloud.com:1521/gcb643e5b4a5586_dpraxhy2bgcmpyzq_low.adb.oraclecloud.com?wallet_location=/Users/jonny/Downloads/wallet"```

> However, support for providing a wallet for the Oracle datasource is not in budibase (we would need to test the feasibility of doing that). 
> I also tried a self-hosted budibase installation where I copied the wallet into the inside of the container and tried pointing the connection string to that location but that workaround didn't work.

Hopefully this helps!
